spec/classes/init_spec.rb

```diff
@@ -1,91 +1,106 @@
-          :client_package => 'custom-gluster-client',
-          :version        => '3.1.4',
-          :repo           => false,
-        )
+      context 'with all defaults' do
+        it { should contain_class('gluster') }
+        it { should contain_class('gluster::params') }
+        it { should compile.with_all_deps }
+
+        it 'should include classes' do
+          should contain_class('gluster::install')
+          should contain_class('gluster::service')
+        end
+        it 'should create gluster::install' do
+          should create_class('gluster::install').with(
+            server: true,
+            server_package: 'glusterfs',
+            client: true,
+            client_package: 'glusterfs-fuse',
+            version: 'LATEST',
+            repo: true,
+          )
+        end
+        it 'should manage the Gluster service' do
+          should create_class('gluster::service').with(
+            ensure: true,
+          )
+        end
       end
-      it 'should manage the Gluster service' do
-        should create_class('gluster::service').with(
-          :ensure => true,
-        )
+      context 'specific version and package names defined' do
+        let :params do {
+          server_package: 'custom-gluster-server',
+          client_package: 'custom-gluster-client',
+          version: '3.1.4',
+          repo: false,
+        }
+        end
+        it 'should create gluster::install' do
+          should create_class('gluster::install').with(
+            server: true,
+            server_package: 'custom-gluster-server',
+            client: true,
+            client_package: 'custom-gluster-client',
+            version: '3.1.4',
+            repo: false,
+          )
+        end
+        it 'should manage the Gluster service' do
+          should create_class('gluster::service').with(
+            ensure: true,
+          )
+        end
+        it 'should install custom-gluster-client and custom-gluster-server' do
+          should create_package('custom-gluster-client')
+          should create_package('custom-gluster-server')
+        end
       end
-    end
 
-    context 'when volumes defined' do
-      let :params do
-        { :volumes =>
+      context 'when volumes defined' do
+        let :facts do
+          super().merge(
+            gluster_binary: '/sbin/gluster',
+            gluster_peer_list: 'example1,example2',
+            gluster_volume_list: 'gl1.example.com:/glusterfs/backup,gl2.example.com:/glusterfs/backup'
+          )
+        end
+        let :params do
           {
-            'data1' => {
-              'replica' => 2,
-              'bricks'  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
-              'options' => ['server.allow-insecure: on'],
+            volumes:
+            {
+              'data1' => {
+                'replica' => 2,
+                'bricks'  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
```

Comment:
o_<

does gluster even support just two bricks??
isn't that a guaranteed case for split-brain?